_posts/2024-01-17-geospatial-techno.md

```diff
@@ -0,0 +1,82 @@
+- The "Modules" tab, provides the status of the various modules installed on the server.
+- The "System Status" tab, provides extra information about the system environment GeoServer is running in.
+
+#### Status Field Descriptions
+This page describes the current status indicators:
+- Data directory: It shows the path to the GeoServer data directory. The procedure for setting the location of the GeoServer data directory is dependent on the type of GeoServer installation. When running a GeoServer Web Archive inside a servlet container, the data directory can be specified in several ways. The recommended method is to set a servlet context parameter. To specify the data directory using a servlet context parameter, create the "context-param" element in the "WEB-INF/web.xml" file for the GeoServer application. After you change the path of the data directory, log in to GeoServer again. Now you can see the new path of the data directory.
+- Locks: When using Transactional Web Feature Service (WFS-T), clients can edit feature types. To prevent data corruption, GeoServer locks the data during transactions. If the number is greater than one, there are active transactions. Clicking "Free Locks" resets a hung editing session and removes any abandoned locks.
+- Connections: This shows you the number of vector data store connections. Vector data stores are repositories configured for the persistence of features.
+- Memory Usage: This shows you how much memory GeoServer is using. You can manually run the garbage collector by clicking the Free Memory button, so it cleans up memory marked for deletion.
+- JVM Version: This is the version of the "Java Virtual Machine" that the GeoServer is using.
+- Java Rendering Engine: It shows the rendering engine used for vector operations.
+- Available Fonts: This is a list of the fonts seen by the GeoServer. Fonts are useful to render labels for spatial features. Selecting the link will show the full list. To add custom fonts to the GeoServer, first, you have to download your favorite fonts from the web, then copy them to the "Java installation folder\jre\lib\fonts". After restarting the Apache Tomcat software, the new fonts will be added to the Available Fonts list.
+
+The Java Advanced Imaging(JAI) is an advanced image manipulation library that reduces the need to add memory to the server. This extension is available for Windows (32-bit version of Java), Linux and Solaris (32 and 64-bit versions). When properly installed, it displays “true” and makes WCS and WMS performance faster and more efficient.
+
+The JAI parameters of GeoServer have the following titles:
+Native JAI , Native JAI ImageIO , JAI Maximum Memory , JAI Memory Usage , JAI Memory Threshold , Number of JAI Tile Threads , JAI Tile Thread Priority
+
+For more information to know how to install JAI, please visit the [link](https://docs.geoserver.geo-solutions.it/edu/en/install_run/jai_io_install.html)
```

Comment:
This is outdated, JAI is no longer available to be installed.

Reply:
I have provided a short explanation to introduce JAI to the users in different versions of GeoServer. Now, which option do you think is better?
1. Should I mention at the end of the JAI topic that the new GeoServer versions no longer support this issue?
2. In general, should I exclude the JAI-related issue from the content?

Reply:
It has not been a great idea to install the native libraries for JAI for over a decade since they fell out of maintenance. 

Indeed as the JIT compiler included the performance difference between pure java and the native libraries got smaller and smaller and it has not been worth the loss of stability associated with the use of native code.

Reply:
Java 11 no longer supports the plugin system used for native JAI installation. It is not GeoServer that changed but java.

It is much better to exclude JAI related issues from this content as they are no longer relevant